pkg/platform/update-api/platform.go

```diff
@@ -0,0 +1,163 @@
+		Timeout: 10 * time.Second,
+	}}, nil
+}
+
+type statusResponse struct {
+	osVersion *semver.Version
+}
+
+func (sr *statusResponse) OK() bool {
+	// Bottlerocket OS version needs to be at least 0.4.1 to support the Update API
+	constraint, _ := semver.NewConstraint(">= 0.4.1")
+	return constraint.Check(sr.osVersion)
+}
+
+// Try to determine if the Bottlerocket version at least 0.4.1
+func (p Platform) Status() (platform.Status, error) {
+	response, err := p.httpClient.Get("http://unix" + "/os")
+	if err != nil {
+		p.log.WithError(err).Error("error when trying to GET '/os'")
+		return nil, err
+	}
+	var osInfo interface{}
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(body, osInfo)
+	if err != nil {
+		return nil, err
+	}
+	// Assert 'version_id' in the OS info json output as string
+	osVersion, err := semver.NewVersion(osInfo.(map[string]interface{})["version_id"].(string))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse 'version_id' field as semver")
+	}
+	return &statusResponse{osVersion: osVersion}, nil
+}
+
+type listAvailableResponse struct {
+	chosenUpdate *UpdateImage
+}
+
+func (lar *listAvailableResponse) Updates() []platform.Update {
+	if lar.chosenUpdate == nil {
+		return nil
+	}
+	updates := make([]platform.Update, 1)
+	updates[0] = lar.chosenUpdate
+	return updates
+}
+
+func (p Platform) ListAvailable() (platform.Available, error) {
+	p.log.Debug("fetching list of available updates")
+
+	// Refresh list of updates and check if there are any available
+	err := postAction(p.httpClient, "/actions/refresh-updates")
+	if err != nil {
+		return nil, err
+	}
+
+	updateStatus, err := getUpdateStatus(p.httpClient)
+	if updateStatus.MostRecentCommand.CmdType == refresh && updateStatus.MostRecentCommand.CmdStatus == Success {
+		// If there is no available update to update to
+		if updateStatus.ChosenUpdate == nil {
+			return &listAvailableResponse{chosenUpdate: nil}, err
+		}
+		return &listAvailableResponse{chosenUpdate: updateStatus.ChosenUpdate}, err
+	} else {
+		return &listAvailableResponse{chosenUpdate: nil}, errors.New("failed to refresh updates or update action performed out of band")
+	}
+}
+
+func (p Platform) Prepare(target platform.Update) error {
+	updateStatus, err := getUpdateStatus(p.httpClient)
+	if err != nil {
+		return err
+	}
+	if updateStatus.UpdateState != Available && updateStatus.UpdateState != Staged {
+		return errors.Errorf("unexpected update state: %s, expecting state to be 'Available' or 'Staged'. update action performed out of band?", updateStatus.UpdateState)
+	}
+
+	// Download the update and apply it to the inactive partition
+	err = postAction(p.httpClient, "/actions/prepare-update")
+	if err != nil {
+		return err
+	}
+
+	updateStatus, err = getUpdateStatus(p.httpClient)
+	if updateStatus.MostRecentCommand.CmdType != prepare || updateStatus.MostRecentCommand.CmdStatus != Success {
```

Comment:
You've got a few places where the only field of `updateStatus` you're accessing is `MostRecentCommand`, and it's pretty long.  Might be worth simplifying to:

```go
command, err := client.GetMostRecentCommand()
```